serial/serialposix.py

```diff
@@ -707,6 +712,21 @@ def send_break(self, duration=0.25):
             raise PortNotOpenError()
         termios.tcsendbreak(self.fd, int(duration / 0.25))
 
+    def _update_rts_and_dtr_state(self):
```

Comment:
It looks like `._update_rts_and_dtr_state()` can supersede the functionality of both `._update_rts_state()` and `._update_dtr_state()`. Is that an accurate assessment?

If so, please eliminate both of the individual update methods in favor of this one. That may reduce the complexity of the code in `.open()`:

```python
if (not self._rtscts) or (not self._dsrdtr):
    self._update_rts_and_dtr_state()
```